0.3.2.8: fails to build with error: 'VK_LAYER_EXPORT' does not name a type

using
glslang: 12.0.0
vulkan: 1.3.241
```
FAILED: src/libvkbasalt.so.p/basalt.cpp.o 
x86_64-pc-linux-gnu-c++ -Isrc/libvkbasalt.so.p -Isrc -I../vkBasalt-0.3.2.8/src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++2a -O0 -g -march=native -O2 -pipe -march=native -O2 -fPIC -MD -MQ src/libvkbasalt.so.p/basalt.cpp.o -MF src/libvkbasalt.so.p/basalt.cpp.o.d -o src/libvkbasalt.so.p/basalt.cpp.o -c ../vkBasalt-0.3.2.8/src/basalt.cpp
../vkBasalt-0.3.2.8/src/basalt.cpp:70:5: error: 'VK_LAYER_EXPORT' does not name a type
   70 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_CreateInstance(const VkInstanceCreateInfo*  pCreateInfo,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:136:5: error: 'VK_LAYER_EXPORT' does not name a type
  136 |     VK_LAYER_EXPORT void VKAPI_CALL vkBasalt_DestroyInstance(VkInstance instance, const VkAllocationCallbacks* pAllocator)
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:151:5: error: 'VK_LAYER_EXPORT' does not name a type
  151 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_CreateDevice(VkPhysicalDevice             physicalDevice,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:286:5: error: 'VK_LAYER_EXPORT' does not name a type
  286 |     VK_LAYER_EXPORT void VKAPI_CALL vkBasalt_DestroyDevice(VkDevice device, const VkAllocationCallbacks* pAllocator)
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:750:5: error: 'VK_LAYER_EXPORT' does not name a type
  750 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateInstanceLayerProperties(uint32_t* pPropertyCount, VkLayerProperties* pProperties)
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:766:5: error: 'VK_LAYER_EXPORT' does not name a type
  766 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateDeviceLayerProperties(VkPhysicalDevice   physicalDevice,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:773:5: error: 'VK_LAYER_EXPORT' does not name a type
  773 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateInstanceExtensionProperties(const char*            pLayerName,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:790:5: error: 'VK_LAYER_EXPORT' does not name a type
  790 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateDeviceExtensionProperties(VkPhysicalDevice       physicalDevice,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:820:5: error: 'VK_LAYER_EXPORT' does not name a type
  820 |     VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName);
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:821:5: error: 'VK_LAYER_EXPORT' does not name a type
  821 |     VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName);
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:860:5: error: 'VK_LAYER_EXPORT' does not name a type
  860 |     VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName)
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:875:5: error: 'VK_LAYER_EXPORT' does not name a type
  875 |     VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName)
      |     ^~~~~~~~~~~~~~~
```
full build log: [vkbasalt-0.3.2.8.log.txt](https://github.com/DadSchoorse/vkBasalt/files/10828612/vkbasalt-0.3.2.8.log.txt)

--- a/src/basalt.cpp
+++ b/src/basalt.cpp
@@ -41,6 +41,12 @@
 
 #define VKBASALT_NAME "VK_LAYER_VKBASALT_post_processing"
 
+#if defined(__GNUC__) && __GNUC__ >= 4
+#define VK_BASALT_EXPORT __attribute__((visibility("default")))
+#else
+#error "Unsupported platform!"
+#endif
+
 namespace vkBasalt
 {
     std::shared_ptr<Config> pConfig = nullptr;
@@ -67,9 +73,9 @@ namespace vkBasalt
         return *(void**) inst;
     }
 
-    VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_CreateInstance(const VkInstanceCreateInfo*  pCreateInfo,
-                                                                const VkAllocationCallbacks* pAllocator,
-                                                                VkInstance*                  pInstance)
+    VK_BASALT_EXPORT VkResult VKAPI_CALL vkBasalt_CreateInstance(const VkInstanceCreateInfo*  pCreateInfo,
+                                                                 const VkAllocationCallbacks* pAllocator,
+                                                                 VkInstance*                  pInstance)
     {
         VkLayerInstanceCreateInfo* layerCreateInfo = (VkLayerInstanceCreateInfo*) pCreateInfo->pNext;
 
@@ -133,7 +139,7 @@ namespace vkBasalt
         return ret;
     }
 
-    VK_LAYER_EXPORT void VKAPI_CALL vkBasalt_DestroyInstance(VkInstance instance, const VkAllocationCallbacks* pAllocator)
+    VK_BASALT_EXPORT void VKAPI_CALL vkBasalt_DestroyInstance(VkInstance instance, const VkAllocationCallbacks* pAllocator)
     {
         scoped_lock l(globalLock);
 
@@ -148,10 +154,10 @@ namespace vkBasalt
         instanceVersionMap.erase(GetKey(instance));
     }
 
-    VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_CreateDevice(VkPhysicalDevice             physicalDevice,
-                                                              const VkDeviceCreateInfo*    pCreateInfo,
-                                                              const VkAllocationCallbacks* pAllocator,
-                                                              VkDevice*                    pDevice)
+    VK_BASALT_EXPORT VkResult VKAPI_CALL vkBasalt_CreateDevice(VkPhysicalDevice             physicalDevice,
+                                                               const VkDeviceCreateInfo*    pCreateInfo,
+                                                               const VkAllocationCallbacks* pAllocator,
+                                                               VkDevice*                    pDevice)
     {
         scoped_lock l(globalLock);
         Logger::trace("vkCreateDevice");
@@ -283,7 +289,7 @@ namespace vkBasalt
         return VK_SUCCESS;
     }
 
-    VK_LAYER_EXPORT void VKAPI_CALL vkBasalt_DestroyDevice(VkDevice device, const VkAllocationCallbacks* pAllocator)
+    VK_BASALT_EXPORT void VKAPI_CALL vkBasalt_DestroyDevice(VkDevice device, const VkAllocationCallbacks* pAllocator)
     {
         scoped_lock l(globalLock);
 
@@ -747,7 +753,7 @@ namespace vkBasalt
     ///////////////////////////////////////////////////////////////////////////////////////////
     // Enumeration function
 
-    VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateInstanceLayerProperties(uint32_t* pPropertyCount, VkLayerProperties* pProperties)
+    VK_BASALT_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateInstanceLayerProperties(uint32_t* pPropertyCount, VkLayerProperties* pProperties)
     {
         if (pPropertyCount)
             *pPropertyCount = 1;
@@ -763,16 +769,16 @@ namespace vkBasalt
         return VK_SUCCESS;
     }
 
-    VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateDeviceLayerProperties(VkPhysicalDevice   physicalDevice,
-                                                                                uint32_t*          pPropertyCount,
-                                                                                VkLayerProperties* pProperties)
+    VK_BASALT_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateDeviceLayerProperties(VkPhysicalDevice   physicalDevice,
+                                                                                 uint32_t*          pPropertyCount,
+                                                                                 VkLayerProperties* pProperties)
     {
         return vkBasalt_EnumerateInstanceLayerProperties(pPropertyCount, pProperties);
     }
 
-    VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateInstanceExtensionProperties(const char*            pLayerName,
-                                                                                      uint32_t*              pPropertyCount,
-                                                                                      VkExtensionProperties* pProperties)
+    VK_BASALT_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateInstanceExtensionProperties(const char*            pLayerName,
+                                                                                       uint32_t*              pPropertyCount,
+                                                                                       VkExtensionProperties* pProperties)
     {
         if (pLayerName == NULL || std::strcmp(pLayerName, VKBASALT_NAME))
         {
@@ -787,10 +793,10 @@ namespace vkBasalt
         return VK_SUCCESS;
     }
 
-    VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateDeviceExtensionProperties(VkPhysicalDevice       physicalDevice,
-                                                                                    const char*            pLayerName,
-                                                                                    uint32_t*              pPropertyCount,
-                                                                                    VkExtensionProperties* pProperties)
+    VK_BASALT_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateDeviceExtensionProperties(VkPhysicalDevice       physicalDevice,
+                                                                                     const char*            pLayerName,
+                                                                                     uint32_t*              pPropertyCount,
+                                                                                     VkExtensionProperties* pProperties)
     {
         // pass through any queries that aren't to us
         if (pLayerName == NULL || std::strcmp(pLayerName, VKBASALT_NAME))
@@ -817,8 +823,8 @@ namespace vkBasalt
 extern "C"
 { // these are the entry points for the layer, so they need to be c-linkeable
 
-    VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName);
-    VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName);
+    VK_BASALT_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName);
+    VK_BASALT_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName);
 
 #define GETPROCADDR(func) \
     if (!std::strcmp(pName, "vk" #func)) \
@@ -857,7 +863,7 @@ extern "C"
         GETPROCADDR(BindImageMemory); \
     }
 
-    VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName)
+    VK_BASALT_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName)
     {
         if (vkBasalt::pConfig == nullptr)
         {
@@ -872,7 +878,7 @@ extern "C"
         }
     }
 
-    VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName)
+    VK_BASALT_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName)
     {
         if (vkBasalt::pConfig == nullptr)
         {

--- a/src/meson.build
+++ b/src/meson.build
@@ -56,5 +56,6 @@ shared_library(meson.project_name().to_lower(),
     link_with: [keyboard_input_x11_lib],
     include_directories : vkBasalt_include_path,
     dependencies : [x11_dep, reshade_dep],
+    gnu_symbol_visibility: 'hidden',
     install : true,
     install_dir : lib_dir)

